File: dis-database/src/main/java/com/atlas/dis/entity/MonsterDrop.java
package com.atlas.dis.entity;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table
public class MonsterDrop implements Serializable {
   private static final long serialVersionUID = 1L;

   @Id
   @GeneratedValue(strategy=GenerationType.IDENTITY)
   private Integer id;

   @Column(nullable = false)
   private Integer monsterId;

   @Column(nullable = false)
   private Integer itemId;

   @Column(nullable = false)
   private Integer minimumQuantity;

   @Column(nullable = false)
   private Integer maximumQuantity;

   @Column(nullable = false)
   private Integer chance;

   public MonsterDrop() {
   }

   public Integer getId() {
      return id;
   }

   public void setId(Integer id) {
      this.id = id;
   }

   public Integer getMonsterId() {
      return monsterId;
   }

   public void setMonsterId(Integer monsterId) {
      this.monsterId = monsterId;
   }

   public Integer getItemId() {
      return itemId;
   }

   public void setItemId(Integer itemId) {
      this.itemId = itemId;
   }

   public Integer getMinimumQuantity() {
      return minimumQuantity;
   }

   public void setMinimumQuantity(Integer minimumQuantity) {
      this.minimumQuantity = minimumQuantity;
   }

   public Integer getMaximumQuantity() {
      return maximumQuantity;
   }

   public void setMaximumQuantity(Integer maximumQuantity) {
      this.maximumQuantity = maximumQuantity;
   }

   public Integer getChance() {
      return chance;
   }

   public void setChance(Integer chance) {
      this.chance = chance;
   }
}

File: atlas.com/dis/domain/monster_drop.go
package domain

type MonsterDrop struct {
	id              uint32
	monsterId       uint32
	itemId          uint32
	minimumQuantity uint32
	maximumQuantity uint32
	chance          uint32
}

func (d MonsterDrop) MonsterId() uint32 {
	return d.monsterId
}

func (d MonsterDrop) ItemId() uint32 {
	return d.itemId
}

func (d MonsterDrop) MinimumQuantity() uint32 {
	return d.minimumQuantity
}

func (d MonsterDrop) MaximumQuantity() uint32 {
	return d.maximumQuantity
}

func (d MonsterDrop) Chance() uint32 {
	return d.chance
}

func (d MonsterDrop) Id() uint32 {
	return d.id
}

type monsterDropBuilder struct {
	id              uint32
	monsterId       uint32
	itemId          uint32
	minimumQuantity uint32
	maximumQuantity uint32
	chance          uint32
}

func NewMonsterDropBuilder(id uint32) *monsterDropBuilder {
	return &monsterDropBuilder{id: id}
}

func (m *monsterDropBuilder) SetMonsterId(monsterId uint32) *monsterDropBuilder {
	m.monsterId = monsterId
	return m
}

func (m *monsterDropBuilder) SetItemId(itemId uint32) *monsterDropBuilder {
	m.itemId = itemId
	return m
}

func (m *monsterDropBuilder) SetMinimumQuantity(minimumQuantity uint32) *monsterDropBuilder {
	m.minimumQuantity = minimumQuantity
	return m
}

func (m *monsterDropBuilder) SetMaximumQuantity(maximumQuantity uint32) *monsterDropBuilder {
	m.maximumQuantity = maximumQuantity
	return m
}

func (m *monsterDropBuilder) SetChance(chance uint32) *monsterDropBuilder {
	m.chance = chance
	return m
}

func (m *monsterDropBuilder) Build() MonsterDrop {
	return MonsterDrop{
		id:              m.id,
		monsterId:       m.monsterId,
		itemId:          m.itemId,
		minimumQuantity: m.minimumQuantity,
		maximumQuantity: m.maximumQuantity,
		chance:          m.chance,
	}
}
